internal/operators/ocs/validations.go

```diff
@@ -115,24 +114,30 @@ func (o *operator) computeNodeResourceUtil(host *models.Host, nodeInfo *resource
 		nodeInfo.missingInventory = true
 		status = "Missing Inventory in some of the hosts"
 		return status, errors.New("Missing Inventory in some of the hosts ") // to indicate that inventory is empty and the ValidationStatus must be Pending
-	} else if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+	}
+	inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+	if err != nil {
 		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
 		return status, err
 	}
 
-	disks := getValidDiskCount(inventory.Disks)
-	if disks > 1 { // OCS must use only the non-boot disks
-		requiredDiskCPU := int64(disks-1) * o.config.OCSRequiredDiskCPUCount
-		requiredDiskRAM := int64(disks-1) * o.config.OCSRequiredDiskRAMGiB
+	diskCount, err := getValidDiskCount(inventory.Disks, host.InstallationDiskID)
+	if err != nil {
+		return err.Error(), err
+	}
+
+	if diskCount > 0 {
+		requiredDiskCPU := diskCount * o.config.OCSRequiredDiskCPUCount
+		requiredDiskRAM := diskCount * o.config.OCSRequiredDiskRAMGiB
 
 		if inventory.CPU.Count < requiredDiskCPU || inventory.Memory.UsableBytes < conversions.GibToBytes(requiredDiskRAM) {
-			status = fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " GiB RAM")
+			status = fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", diskCount, " non-bootable disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " GiB RAM")
```

Comment:
Please remove non-bootable from the message

Reply:
done